controllers/cluster_create.go

```diff
@@ -767,11 +767,20 @@ func (r *ClusterReconciler) createOrPatchPodMonitor(ctx context.Context, cluster
 
 	// Checking for the PodMonitor resource in the cluster
 	havePodMonitor, err := utils.PodMonitorExist(r.DiscoveryClient)
-	if err != nil || !havePodMonitor {
-		contextLogger.Debug("Kind PodMonitor not detected", "err", err)
+	if err != nil {
 		return err
 	}
 
+	// If the PodMonitor does not exist then check in cluster if it is present, log the  warning.
+	// If it is not enabled in the cluster then no need to continue.
+	if !havePodMonitor {
+		if cluster.IsPodMonitorEnabled() {
+			contextLogger.Warning("Kind PodMonitor not detected, " +
+				"Please create after enabling PodMonitor in cluster")
+		}
+		return nil
```

Comment:
this logic is completely wrong: if we don't have pod monitor, and it is enabled, log and do nothing?!!!